api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -40,25 +53,38 @@
   private static final int PROJECT_BILLING_ID_SIZE = 8;
   @VisibleForTesting static final Duration CREATING_TIMEOUT = Duration.ofMinutes(60);
   @VisibleForTesting static final Duration ASSIGNING_TIMEOUT = Duration.ofMinutes(10);
+  private static final Measure.MeasureLong BUFFER_ENTRY_MEASUREMENT = Measure.MeasureLong.create(
+      "buffer_entries",
+      "The number of billing projects in the buffer",
+      "projects");
+  private static final BucketBoundaries PROJECT_BOUNDARIES = BucketBoundaries.create(
+      Lists.newArrayList(0d, 10d, 20d, 50d, 200d));
   private static ImmutableMap<BufferEntryStatus, Duration> statusToGracePeriod =
       ImmutableMap.of(
           BufferEntryStatus.CREATING, CREATING_TIMEOUT,
           BufferEntryStatus.ASSIGNING, ASSIGNING_TIMEOUT);
   private final BillingProjectBufferEntryDao billingProjectBufferEntryDao;
   private final Clock clock;
   private final FireCloudService fireCloudService;
+  private final MonitoringService monitoringService;
   private final Provider<WorkbenchConfig> workbenchConfigProvider;
 
   @Autowired
   public BillingProjectBufferService(
       BillingProjectBufferEntryDao billingProjectBufferEntryDao,
       Clock clock,
       FireCloudService fireCloudService,
+      MonitoringService monitoringService,
```

Comment:
A pattern that has worked well in the auditing system was to introduce site-specific adapter services to handle the event construction logic for each class being instrumented. This minimized the intrusiveness of the auditing system and also insulated the code being monitored from details of the actual system.

Reply:
Let's sync up on this tomorrow. I remember seeing that but didn't understand it as well as I could have, so pairing on the impl sounds good.